climpred/relative_entropy.py

```diff
@@ -0,0 +1,343 @@
+                                              - LENS: member, time
+                                              - PM_control: initialization
+        anomaly_data (bool): Input data is anomaly alread. Default: False.
+        neofs (int): number of EOFs to use. Default: initialized.member.size.
+        curv (bool): if curvilinear grids are provided disables EOF weights.
+        ntime (int): number of timesteps calculated.
+        detrend_by_control_uninitialized (bool): Default: True
+        nmember_control (int): number of members created from bootstrapping from control
+
+    Returns:
+        rel_ent (pd.DataFrame): relative entropy
+
+    """
+    # Defaults
+    if neofs is None:
+        neofs = initialized.member.size
+    if ntime is None:
+        ntime = initialized.time.size
+
+    # case if you only submit control with dim time, PM case
+    if ('ensemble' not in control.dims) and ('member' not in control.dims):
+        control_uninitialized = xr.concat([_bootstrap_dim(control, initialized.time.size, dim='member', dim_label=np.arange(
+            nmember_control)) for _ in range(initialized.initialization.size)], dim='initialization')
+        if isinstance(control_uninitialized, xr.DataArray):
+            control_uninitialized['initialization'] = initialized.initialization.values
+        elif isinstance(control_uninitialized, xr.Dataset):
+            control_uninitialized = control_uninitialized.assign(
+                initialization=initialized.initialization.values)
+
+    # case if you submit control with dim time and member, LENS case
+    elif 'member' in control.dims:
+        control_uninitialized = _bootstrap_dim(
+            control, initialized.time.size, time_dim='time', dim='initialization', dim_label=list(initialized.initialization.values))
+        #print('added member')
+
+    # initialized and control_uninitialized are allowed to have different dims
+    # as I need more members to sample my control distribution properly
+    # ToDo: understand this more
+    if initialized.dims != control_uninitialized.dims:
+        print(initialized.dims)
+        print(control_uninitialized.dims)
+        # raise ValueError('init and uninit must have same dims')
+        warnings.warn(
+            "Warning: initialized and control_uninitialized have different coords.")
+
+    if isinstance(control_uninitialized, xr.Dataset):
+        control_uninitialized = control_uninitialized.to_array().squeeze()
+    if isinstance(initialized, xr.Dataset):
+        initialized = initialized.to_array().squeeze()
+
+    # detrend
+    #non_spatial_dims = set(control_uninitialized.dims).intersection(['time', 'initialization','member'])
+    non_spatial_dims = set(control_uninitialized.dims).intersection(
+        ['initialization', 'member'])
+    non_spatial_dims = list(non_spatial_dims)
+    if not anomaly_data:  # if ds, control are raw values
+        if detrend_by_control_unitialized:
```

Comment:
Why is this an if statement if there is no alternative to generating anomalies? I.e., there's no `elif` or `else` to this.

Reply:
rm if